src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +834,137 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        /// <summary>
+        /// Capture the set of compilation options to allow a compilation 
+        /// to be reconstructed from the pdb
+        /// </summary>
+        private void EmbedCompilationOptions(CommonPEModuleBuilder module)
+        {
+
+            var builder = new BlobBuilder();
+
+            var compilerVersion = typeof(Compilation).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
+            WriteValue("compiler-version", compilerVersion);
+
+            if (module.EmitOptions.FallbackSourceFileEncoding != null)
+            {
+                WriteValue("fallback-encoding", module.EmitOptions.FallbackSourceFileEncoding.WebName);
+            }
+
+            if (module.EmitOptions.DefaultSourceFileEncoding != null)
+            {
+                WriteValue("default-encoding", module.EmitOptions.DefaultSourceFileEncoding.WebName);
+            }
+
+            int portabilityPolicy = 0;
+            if (module.CommonCompilation.Options.AssemblyIdentityComparer is DesktopAssemblyIdentityComparer identityComparer)
+            {
+                portabilityPolicy |= identityComparer.PortabilityPolicy.SuppressSilverlightLibraryAssembliesPortability ? 0b1 : 0;
+                portabilityPolicy |= identityComparer.PortabilityPolicy.SuppressSilverlightPlatformAssembliesPortability ? 0b10 : 0;
+            }
+
+            WriteValue("portability-policy", portabilityPolicy.ToString());
+
+            var runtimeVersion = typeof(object).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
+            WriteValue("runtime-version", runtimeVersion);
+
+            var optimization = module.CommonCompilation.Options.OptimizationLevel == OptimizationLevel.Debug
+                ? (module.CommonCompilation.Options.DebugPlusMode ? "debug+" : "debug")
+                : "release";
```

Comment:
The logic of converting `OptimizationLevel` to a `string` is duplicated in two places. If we keep this approach versus just using `ToString` we need to centralize this logic.

Reply:
It should only be in the test validating this, and the implementation that writes it.

Reply:
Still that's two places. Should add a function to make this more maintainable.

Reply:
Note: the logic exists two places in production code, not test code: `SerializePdbEmbeddedCompilationOptions` and here.

Reply:
It should be removed from `CSharpCompilation.SerializePdbEmbeddedCompilationOptions`

I don't mind adding a helper either. If we decide to keep these as one option serialized then I'll do that.

Reply:
Added `OptimizationLevelFacts.ToPdbSerializedString` and `OptimizationLevelFacts.TryParsePdbSerializedString` after discussing offline

Reply:
@ryzngard still see the same code. Did u push this change yet?

Reply:
apparently my push actually failed from some error o.O pushed again and succeeded.